main/src/main/scala/sbt/Main.scala

```diff
@@ -33,13 +33,24 @@ import xsbti.compile.CompilerCache
 import scala.annotation.tailrec
 import scala.concurrent.ExecutionContext
 import scala.util.control.NonFatal
+import xsbti.AppProvider
 
 /** This class is the entry point for sbt. */
 final class xMain extends xsbti.AppMain {
   def run(configuration: xsbti.AppConfiguration): xsbti.MainResult =
     new XMainConfiguration().run("xMain", configuration)
 }
 private[sbt] object xMain {
+  private[sbt] def dealiasBaseDirectory(config: xsbti.AppConfiguration): xsbti.AppConfiguration = {
+    val dealiasedBase = config.baseDirectory.getCanonicalFile
+    if (config.baseDirectory == dealiasedBase) config
+    else
+      new xsbti.AppConfiguration {
+        override def arguments: Array[String] = config.arguments()
+        override val baseDirectory: File = dealiasedBase
+        override def provider: AppProvider = config.provider()
+      }
+  }
```

Comment:
One concern I have about dealiasing the base directory is the discrepancies it might introduce between external editors. For example between VS Code and sbt server there's a frequent communication about which files are opened, if the files is under sbt's jurisdiction, and if the file was saved by the user etc. If VS Code sends an aliased path like `/tmp/hello/Hello.scala`, but we internally track that as `/private/tmp/hello/Hello.scala` wouldn't that cause a mismatch? /cc @adpi2

Reply:
Or more simpler example might be terminals supporting click-to-open when it finds a full path in compiler errors.

Reply:
That is a valid concern. I don't know how editors / lsp clients work with files. The heart of this change is that, in my opinion, `baseDirectory` and `appConfiguration.value.baseDirectory` should always return the same file. That is the motivation for this change. My belief is that prior to the introduction of `rootPath`, pretty much all paths generated by sbt were canonicalized with respect to the base directory due to the dealising in `Load`. Just now, I audited all of the calls to `appConfiguration.value` with sbt itself and there was only one place I could find where `baseDirectory` was used and not canonicalized: https://github.com/sbt/sbt/blob/e323f1f713251968b3d4ecda8a7f0ed52593713a/main/src/main/scala/sbt/internal/DefaultBackgroundJobService.scala#L504.

Not making this change seems more likely to me to cause problems because right now we have an inconsistency and downstream users are forced to either canonicalize `AppConfiguration.baseDirectory` or potentially have inconsistency when comparing sbt generated paths that use `baseDirectory`.

Reply:
Note also that this only matters if the baseDirectory of the project is actually a symlink. I bet that outside of scripted and sbt tests, that is quite rare.

Reply:
As a side not, it would be kind of nice if sbt only used paths relative to the base directory. Then there would be no need for `rootPaths` to convert paths to absolute paths. It would also make logging of path names nicer. Global directories like the coursier cache could be handled with symlinks in the project directory. There are obviously some potential issues with this approach (and would probably need to be an sbt 2.0 feature) but it's interesting to think about.

Reply:
> My belief is that prior to the introduction of `rootPath`, pretty much all paths generated by sbt were canonicalized with respect to the base directory due to the dealising in `Load`.

ok. So this PR corrects the path handling back to status quo (1.3.x). I guess we'll cross the editor + symlink issue when it materializes then.

Reply:
> If VS Code sends an aliased path like /tmp/hello/Hello.scala, but we internally track that as /private/tmp/hello/Hello.scala wouldn't that cause a mismatch?

What I can tell about BSP is that file paths are going from sbt to client and not the other way around. For instance sbt can tell that the source of target foo are in the /foo/src/main. Then Metals will watch that directory for a change, but when a change happens it will send a compile request of target foo and it will not mention the file.

The only hypothetical mismatch I can think of would be that VSCode works on the aliased path but sbt reports compile error in the dealiased path.